src/Raven.Studio/typescript/models/database/index/indexDefinition.ts

```diff
@@ -311,15 +313,21 @@ class indexDefinition {
 
     addField() {
         const field = indexFieldOptions.empty();
+        
+        field.addCustomAnalyzers(this.customAnalyzers());
+        
         if (this.defaultFieldOptions()) {
             field.parent(this.defaultFieldOptions());
         }
+        
         this.fields.unshift(field);
     }
 
     addDefaultField() {
         const fieldOptions = indexFieldOptions.defaultFieldOptions();
         this.defaultFieldOptions(fieldOptions);
+        
+        this.defaultFieldOptions().addCustomAnalyzers(this.customAnalyzers());
```

Comment:
```
const fieldOptions = indexFieldOptions.defaultFieldOptions();
fieldOptions.addCustomAnalyzers(this.customAnalyzers());
this.defaultFieldOptions(fieldOptions);
```

such order will avoid accessing observables and reduces amount of UI changes.

Reply:
done